sequencers/modeshape-sequencer-video/src/main/java/org/modeshape/sequencer/video/VideoMetadata.java

```diff
@@ -0,0 +1,200 @@
+/*
+ * ModeShape (http://www.modeshape.org)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.modeshape.sequencer.video;
+
+import io.humble.video.Decoder;
+import io.humble.video.Demuxer;
+import io.humble.video.DemuxerStream;
+import io.humble.video.Global;
+import io.humble.video.KeyValueBag;
+import io.humble.video.MediaDescriptor.Type;
+
+import java.io.DataInput;
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Utility for extracting Metadata from video formats.
+ */
+public class VideoMetadata {
+
+    static final String[] MIME_TYPE_STRINGS = { "video/x-matroska",  // mkv
+                                                "video/quicktime",  // mov
+                                                "video/x-msvideo",  // avi
+                                                "video/x-ms-wmv",  // wmv
+                                                "video/x-flv",  // flv
+                                                "video/3gpp",  // 3gpp
+                                                "video/webm",  // webp
+                                                "video/mp4",  // mp4
+                                                "video/ogg"  //ogg
+    };
+
+    private Double duration;
+    private Integer bitrate;
+    private String title;
+    private String comment;
+    private String encoder;
+    private List<StreamMetadata> streams = new ArrayList<StreamMetadata>();
+
+    private InputStream in;
+    private DataInput din;
+
+    /*
+     * Check that given file is supported by this sequencer.
+     */
+    public boolean check() {
+        try {
+            // create a temporary copy from output
+            File fileCopy = File.createTempFile("modeshape-sequencer-video", ".tmp");
+            FileOutputStream fileOutputStream = new FileOutputStream(fileCopy);
+            byte[] b = new byte[1024];
+            while (read(b) != -1) {
+                fileOutputStream.write(b);
+            }
+            fileOutputStream.close();
+
+            final Demuxer container = Demuxer.make();
+            container.open(fileCopy.getAbsolutePath(), null, false, true, null, null);
```

Comment:
Demuxer should be closed when finished, according to
https://github.com/artclarke/humble-video/blob/master/humble-video-noarch/src/main/java/io/humble/video/Demuxer.java#L174